pkg/liveshare/ports.go

```diff
@@ -0,0 +1,135 @@
+package liveshare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/sourcegraph/jsonrpc2"
+)
+
+// Port describes a port exposed by the container.
+type Port struct {
+	SourcePort                       int    `json:"sourcePort"`
+	DestinationPort                  int    `json:"destinationPort"`
+	SessionName                      string `json:"sessionName"`
+	StreamName                       string `json:"streamName"`
+	StreamCondition                  string `json:"streamCondition"`
+	BrowseURL                        string `json:"browseUrl"`
+	IsPublic                         bool   `json:"isPublic"`
+	IsTCPServerConnectionEstablished bool   `json:"isTCPServerConnectionEstablished"`
+	HasTLSHandshakePassed            bool   `json:"hasTLSHandshakePassed"`
+	Privacy                          string `json:"privacy"`
+}
+
+type PortChangeKind string
+
+const (
+	PortChangeKindStart  PortChangeKind = "start"
+	PortChangeKindUpdate PortChangeKind = "update"
+)
+
+type PortUpdate struct {
+	Port        int            `json:"port"`
+	ChangeKind  PortChangeKind `json:"changeKind"`
+	ErrorDetail string         `json:"errorDetail"`
+	StatusCode  int            `json:"statusCode"`
+}
+
+// startSharing tells the Live Share host to start sharing the specified port from the container.
+// The sessionName describes the purpose of the remote port or service.
+// It returns an identifier that can be used to open an SSH channel to the remote port.
+func (s *Session) startSharing(ctx context.Context, sessionName string, port int) (channelID, error) {
+	args := []interface{}{port, sessionName, fmt.Sprintf("http://localhost:%d", port)}
+	errc := make(chan error, 1)
+
+	go func() {
+		startNotification, err := s.WaitForPortNotification(ctx, port, PortChangeKindStart)
```

Comment:
I've made the switch in the latest commit, however I'm pretty new to Go and wanna check in and make sure this won't lead to the scoping issues @markphelps pointed out [here](https://github.com/cli/cli/pull/5270#r818129970), since there doesn't seem to be a way to pass arguments to the anonymous function in Group.Go()

Reply:
The code looks sound to me, because the wait statement ensures that each goroutine is finished with 'port' before the next loop iteration updates it.